src/main/resources/assets/js/app/view/PublicKeysGrid.ts

```diff
@@ -0,0 +1,124 @@
+import {DivEl} from '@enonic/lib-admin-ui/dom/DivEl';
+import {LabelEl} from '@enonic/lib-admin-ui/dom/LabelEl';
+import {AEl} from '@enonic/lib-admin-ui/dom/AEl';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {ConfirmationDialog} from '@enonic/lib-admin-ui/ui/dialog/ConfirmationDialog';
+import {PublicKey} from '../browse/serviceaccount/PublicKey';
+import {User} from '../principal/User';
+import {DeletePublicKeyRequest} from '../../graphql/principal/user/DeletePublicKeyRequest';
+import {UserKeyDetailsDialog} from '../wizard/UserKeyDetailsDialog';
+import {DateHelper} from '@enonic/lib-admin-ui/util/DateHelper';
+
+export class PublicKeysGrid
+    extends DivEl {
+
+    private readonly tbody: DivEl;
+
+    private removeCallback: () => void;
+
+    constructor(className ?: string) {
+        super(className || '');
+
+        this.getEl().addClass('public-keys-grid');
+
+        const publicKeysHeader = new LabelEl(i18n('field.userKeys.grid.title'));
+        publicKeysHeader.addClass('public-keys-label');
+        this.appendChild(publicKeysHeader);
+
+
+        const header = this.createDivEl('public-keys-row', 'rowgroup');
+        header.appendChild(this.createDivEl('public-keys-cell', 'columnheader', i18n('field.userKeys.grid.kid.column')));
+        header.appendChild(this.createDivEl('public-keys-cell', 'columnheader', i18n('field.userKeys.grid.label.column')));
+        header.appendChild(this.createDivEl('public-keys-cell', 'columnheader', i18n('field.userKeys.grid.creationTime.column')));
+        header.appendChild(this.createDivEl('public-keys-cell', 'columnheader', i18n('field.userKeys.grid.actions.column')));
```

Comment:
This needs some refactoring:
* Add separate methods that create a regular cell and a header cell:

```Typescript
private createCell(html: string, role?: string): DivEl {
    return this.createDivEl('public-keys-cell', role || 'cell', html);
}

private createHeaderCell(html: string): DivEl {
    return this.createCell(html, 'columnheader');
}
```
* Create separate methods for creating a header and a regular row:

```Typescript
private createHeader(): DivEl {
      const header = this.createDivEl('public-keys-header', 'gridheader');
      header.appendChild(this.createHeaderCell(i18n('field.userKeys.grid.kid.column')));
      header.appendChild(this.createHeaderCell(i18n('field.userKeys.grid.label.column')));
      header.appendChild(this.createHeaderCell(i18n('field.userKeys.grid.creationTime.column')));
      header.appendChild(this.createHeaderCell(i18n('field.userKeys.grid.actions.column')));

      return header;
}

private createDataRow(publicKey: PublicKey): DivEl {
      const row = this.createDivEl('public-keys-row', 'rowgroup');
      row.appendChild(this.createCell(publicKey.getKid()));
      row.appendChild(this.createCell(publicKey.getLabel()));

      const creationTime = new Date(Date.parse(publicKey.getCreationTime()));
      row.appendChild(this.createCell(DateHelper.formatDateTime(creationTime)));

      return row;
}
```

* Change your code to call these new methods when needed.